bundles/org.openhab.binding.openweathermap/src/main/java/org/openhab/binding/openweathermap/internal/handler/OpenWeatherMapOneCallHistoryHandler.java

```diff
@@ -0,0 +1,309 @@
+ */
+package org.openhab.binding.openweathermap.internal.handler;
+
+import com.google.gson.JsonSyntaxException;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.i18n.TimeZoneProvider;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.openweathermap.internal.config.OpenWeatherMapOnecallHistoryConfiguration;
+import org.openhab.binding.openweathermap.internal.connection.OpenWeatherMapCommunicationException;
+import org.openhab.binding.openweathermap.internal.connection.OpenWeatherMapConfigurationException;
+import org.openhab.binding.openweathermap.internal.connection.OpenWeatherMapConnection;
+import org.openhab.binding.openweathermap.internal.dto.onecallhist.*;
+import org.openhab.binding.openweathermap.internal.dto.onecallhist.OpenWeatherMapOneCallHistAPIData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.eclipse.smarthome.core.library.unit.MetricPrefix.HECTO;
+import static org.eclipse.smarthome.core.library.unit.MetricPrefix.MILLI;
+import static org.eclipse.smarthome.core.library.unit.MetricPrefix.KILO;
+import static org.eclipse.smarthome.core.library.unit.SIUnits.*;
+import static org.eclipse.smarthome.core.library.unit.SmartHomeUnits.*;
+import static org.openhab.binding.openweathermap.internal.OpenWeatherMapBindingConstants.*;
+
+/**
+ * The {@link OpenWeatherMapOneCallHistoryHandler} is responsible for handling commands, which are sent to one of
+ * the channels.
+ *
+ * @author Wolfgang Klimt - Initial contribution
+ */
+@NonNullByDefault
+public class OpenWeatherMapOneCallHistoryHandler extends AbstractOpenWeatherMapHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(OpenWeatherMapOneCallHistoryHandler.class);
+
+    private static final String CHANNEL_GROUP_HOURLY_FORECAST_PREFIX = "historyHours";
+    private static final Pattern CHANNEL_GROUP_HOURLY_FORECAST_PREFIX_PATTERN = Pattern
+            .compile(CHANNEL_GROUP_HOURLY_FORECAST_PREFIX + "([0-9]*)");
+
+    private @Nullable OpenWeatherMapOneCallHistAPIData weatherData;
+
+    // the relative day in history.
+    private int day = 0;
+
+    public OpenWeatherMapOneCallHistoryHandler(Thing thing, final TimeZoneProvider timeZoneProvider) {
+        super(thing, timeZoneProvider);
+    }
+
+    @Override
+    public void initialize() {
+        super.initialize();
+        OpenWeatherMapOnecallHistoryConfiguration config = getConfigAs(
+                OpenWeatherMapOnecallHistoryConfiguration.class);
+        if (config.historyDay <= 0) {
+            logger.error("historyDay value of {} is not supported", config.historyDay);
```

Comment:
```suggestion
            logger.warn("historyDay value of {} is not supported", config.historyDay);
```
error logs are used for when breaking problems. For more information about log levels we use see: https://www.openhab.org/docs/developer/guidelines.html#f-logging

Reply:
Done